autoload/go/util.vim

```diff
@@ -99,7 +99,13 @@ endfunction
 function! go#util#System(str, ...) abort
   let l:shell = &shell
   if !go#util#IsWin() && executable('/bin/sh')
-    let &shell = '/bin/sh'
+    " Having the [t]csh shells set as the login shell results in temporary
+    " file access errors when calling ksh or dash system shells. /bin/sh is
+    " often aliased to these shells on modern Linux and BSD systems including
+    " VoidLinux, Debian and FreeBSD.
+    if match(l:shell, 'csh$') == -1
+      let &shell = '/bin/sh'
+    endif
```

Comment:
> What problems are you having?

The regression introduced by #988 causes the following problems in a *csh, zsh login environment on a system with /bin/sh pointing to dash, ksh or BSD sh.

Problem 1. Running 'vim test.go' results in errors and no template.
Problem 2. Unable to save a modified *.go file. Errors in save process prevents successful write.
Problem 3. Most vim-go commands resulting in calling the sub-shell fail with error.

Example error message produced from 'vim test.go':
`Error detected while processing function <SNR>27_template_autocreate[3]..go#temp
late#create[8]..go#tool#PackageName[2]..go#tool#ExecuteInDir[4]..go#util#env[7].
.go#util#goroot[1]..go#util#System:
line   17:
E484: Can't open file /tmp/vQAaQRG/1
Error detected while processing function <SNR>27_template_autocreate[3]..go#temp
late#create[8]..go#tool#PackageName[2]..go#tool#ExecuteInDir[4]..go#util#env:
line    7:
E171: Missing :endif
Error detected while processing function <SNR>27_template_autocreate:
line    3:
E171: Missing :endif
`